synfig-studio/src/gui/app.cpp

```diff
@@ -1362,15 +1361,68 @@ App::get_default_accel_map()
 
 	return default_accel_map;
 }
+Glib::RefPtr<App> App::instance() {
+	static Glib::RefPtr<studio::App> app_reference = Glib::RefPtr<App>(new App());
+	return app_reference;
+}
 
 /* === M E T H O D S ======================================================= */
+App::App()
+	: Gtk::Application("org.synfig.SynfigStudio", Gio::APPLICATION_HANDLES_OPEN)
+{
+	add_main_option_entry(Gio::Application::OPTION_TYPE_BOOL, "console", 'c', N_("Opens a console that shows some Synfig Studio output"));
+	signal_handle_local_options().connect(sigc::mem_fun(*this, &App::on_handle_local_options));
+}
+
+int App::on_handle_local_options(const Glib::RefPtr<Glib::VariantDict> &options)
+{
+	if (!options)
```

Comment:
It's not intuitive why we are returning -1 here, so I think maybe add a comment here? 
```
To continue, return -1 to let the default option processing continue.
https://developer-old.gnome.org/glibmm/unstable/classGio_1_1Application.html#a46b21ab9629b123b7524fe906290d32b
```